superset/models/core.py

```diff
@@ -767,6 +767,20 @@ def get_sqla_engine(self, schema=None, nullpool=True, user_name=None):
         if DB_CONNECTION_MUTATOR:
             url, params = DB_CONNECTION_MUTATOR(
                 url, params, effective_username, security_manager)
+
+        custom_url_params = None
```

Comment:
Can't this all be handled in the `DB_CONNECTION_MUTATOR`? Currently it feels like you're trying to solve an Airbnb specific problem. In you custom mutator you could probably use [inspect](https://stackoverflow.com/questions/2654113/how-to-get-the-callers-method-name-in-the-called-method) to determine the context in which engine was instantiated.